File: algorithms/dynamic-array/dynamicArray.php
<?php
function dynamicArray($n, $queries) {
  $lastAnswer = 0;
  $arr = [];
  $result = [];
  
  for ($i = 0; $i < $n; $i++) {
    array_push($arr, []);
  }

  foreach ($queries as $q) {
    $queryType = $q[0];
    $x = $q[1];
    $y = $q[2];
    $idx = (($x ^ $lastAnswer) % $y);

    if ($queryType === "1") {
      array_push($arr[$idx], $y);
    } else {
      $lastAnswer = $arr[$idx][$y % count($arr[$idx])];
      array_push($result, $lastAnswer);
    }
  };

  return $result;
}
?>

File: algorithms/dynamic-array/dynamicArray.py
def dynamicArray(n, queries):
  lastAnswer = 0
  arr = [[] for x in range(0, n)]
  result = []

  for q in queries:
    queryType = q[0]
    x = q[1]
    y = q[2]
    idx = ((x ^ lastAnswer) % n)
        
    if (int(queryType) == 1):
      arr[idx].append(y)
    else:
      lastAnswer = arr[idx][y % len(arr[idx])]
      result.append(lastAnswer)
            
  return result